### PR TITLE
VMECIO.save work with all profile types

### PR DESCRIPTION
**Pull request by ddudt**
Allows saving VMEC wout quantities for all types of Equilibrium profiles (pressure vs kinetic, iota vs current). 

Makes a similar change to allow writing DESC input files for Equilibria with any profile type.

### CODE REVIEW

File: desc/io/input_reader.py [hunk 1]
```diff
@@ -12,6 +12,8 @@
 from termcolor import colored
 
 from desc import set_device
+from desc.grid import LinearGrid
```
Comment (dpanici) [inline desc/io/input_reader.py:15]: move to local imports not at top

**Review by dpanici — CHANGES_REQUESTED** [review]
- add test for saving to VMEC with kinetic profile DESC eq

 - move imports to avoid circular import

**Review by f0uriest — DISMISSED** [review]

File: desc/input_reader.py [hunk 2]
```diff
@@ -845,91 +848,77 @@ def descout_to_input(  # noqa: C901 - fxn too complex
...
 
         f.write("\n# fixed-boundary surface shape\n")
+
         # boundary parameters
-        if eq0.sym:
-            for k, (l, m, n) in enumerate(eq0.surface.R_basis.modes):
-                if abs(eq0.Rb_lmn[k]) > 1e-8:
+        if eq.sym:
+            for k, (l, m, n) in enumerate(eq.surface.R_basis.modes):
+                if abs(eq.Rb_lmn[k]) > 1e-8:
```
Comment (dpanici) [inline desc/input_reader.py:904]: Wait I did not realize, the desc output to input only saves boundary coefficients larger than 1e-8?

I think at the least this threshold should be an input parameter, and I would put it lower to like 1e-10 as the default

Comment (dpanici) [reply desc/input_reader.py:904]: #696 not for this PR but for the future making this issue

**Review by dpanici — APPROVED** [review]

File: desc/input_reader.py (hunk 2 above)
Comment (f0uriest) [reply desc/input_reader.py:904]: Iirc you were the one who added this feature and I recommended making it an option at the time

**Review by f0uriest — APPROVED** [review]

**Pull request merged**